test/worker_test.exs

```diff
@@ -130,15 +138,70 @@ defmodule WorkerTest do
          }}
       end)
 
-      Process.sleep(1_000)
+      send(context.crawler, :work)
       assert Process.alive?(context.crawler)
     end
+
+    test "Requests are dropped after 3 attempts", context do
```

Comment:
I'm thinking that there are two options for the worker to know if a request should be sent back for retry:
1. A request key boolean flag (like`retry_if_fail`) that any middleware can set to true, and the worker simply checks if it is true or false
2. The worker checks if a retry middleware is declared. If so, automatically retries failed requests.

I think option 1 is much better, since it isn't rigid, and gives more flexibility in whether a retry should be done and it is on a per-request basis